src/promptflow/promptflow/_sdk/entities/_run.py

```diff
@@ -531,3 +533,27 @@ def _validate_and_return_run_name(run: Union[str, "Run"]) -> str:
         elif isinstance(run, str):
             return run
         raise InvalidRunError(f"Invalid run {run!r}, expected 'str' or 'Run' object but got {type(run)!r}.")
+
+    def _generate_output_path(self, config: Optional[Configuration]) -> Path:
+        config = config or Configuration.get_instance()
+        path = config.get_run_output_path()
+        if path is None:
+            path = Path.home() / PROMPT_FLOW_DIR_NAME / ".runs"
+        else:
+            try:
+                flow_posix_path = self.flow.resolve().as_posix()
+                path = Path(path.replace(FLOW_DIRECTORY_MACRO_IN_CONFIG, self.flow.resolve().as_posix())).resolve()
+                # in case user manually modifies ~/.promptflow/pf.yaml
+                # fall back to default run output path
+                if path.as_posix() == flow_posix_path:
+                    raise Exception(f"{FLOW_DIRECTORY_MACRO_IN_CONFIG!r} is not a valid value.")
+                path.mkdir(parents=True, exist_ok=True)
+            except Exception:  # pylint: disable=broad-except
+                path = Path.home() / PROMPT_FLOW_DIR_NAME / ".runs"
+                warning_message = (
+                    "Got unexpected error when parsing specified output path: "
+                    f"{config.get_run_output_path()!r}; "
+                    f"will use default output path: {path!r} instead."
+                )
+                print_yellow_warning(warning_message)
```

Comment:
use logger.warning, print yellow shall not be used in sdk

Reply:
replaced